django/db/models/sql/query.py

```diff
@@ -1619,6 +1625,8 @@ def names_to_path(self, names, opts, allow_many=True, fail_on_missing=False):
             field = None
             filtered_relation = None
             try:
+                if opts is None:
+                    raise FieldDoesNotExist
```

Comment:
This is only tested in `constraints.tests.CheckConstraintTests.test_validate` and `constraints.tests.UniqueConstraintTests.test_validate_expression_condition`. Can we add a regression test in the `FromEmptyQuery`?